slack-api-client/src/main/java/com/slack/api/methods/impl/AsyncRateLimitExecutor.java

```diff
@@ -60,7 +62,9 @@ public static AsyncRateLimitExecutor getOrCreate(MethodsClientImpl client, Slack
             Map<String, String> params,
             AsyncExecutionSupplier<T> methodsSupplier) {
         String token = params.get("token");
-        final String teamId = token != null ? teamIdCache.lookupOrResolve(token) : null;
+        final String teamId = (token != null
+                && METHOD_NAMES_TO_SKIP_TEAM_ID_CACHE_RESOLUTION.contains(methodName)) ?
```

Comment:
Good catch! This is a bug
```suggestion
                && !METHOD_NAMES_TO_SKIP_TEAM_ID_CACHE_RESOLUTION.contains(methodName)) ?
```